dynamorm/table.py

```diff
@@ -255,12 +255,13 @@ def scan(self, scan_kwargs=None, **kwargs):
             assert len(parts) == 0, "Left over parts after parsing query attr"
 
             op = getattr(attr, op)
+            op_fn = op(value) if value is not None else op()
```

Comment:
This is a strange one, and your test case of `TestModel.scan(baz__not_exists=None)` reads weird to me.  I feel that `TestModel.scan(baz__not_exists=True)` reads better.

It also means that `baz=None` will not work since the eq operator requires an arg.

I think that we should handle this with something like:

```
try:
    filter_expression = op(value)
except TypeError:
    # a TypeError calling our attr op likely means we're invoking exists, not_exists or another op that doesn't take an arg
    # if our value is True then we try to re-call the op function without any arguments, otherwise we bubble it up
    if value is True:
        filter_condition = op()
    else:
        raise
```